google/cloud/storage/bucket.py

```diff
@@ -1787,11 +1790,16 @@ def delete_blobs(
         for blob in blobs:
             try:
                 blob_name = blob
+                generation = None
                 if not isinstance(blob_name, str):
                     blob_name = blob.name
+                    if blob.generation:
+                        generation = blob.generation
```

Comment:
I'm still uncertain whether we should change behavior in this way, though I do agree the current behavior is bad. If we do make this change, can we skip the "take the name from the blob and recreate it" runaround and just run delete on the blob objects directly? This seems unnecessarily convoluted.

Reply:
I understand your concerns. Unfortunately, with the current setup, it will be difficult to bypass "take the name from the blob...". Reasons being (1) params `blobs` can be a list of `str` blob_name OR blob instances, (2) `bucket.delete_blob()` only takes `str` blob_name whereas `blob.delete()` requires a blob instance.

Reply:
As discussed, I'm wondering if adding a flag to the method will better ensure backwards compatibility, but I'm also concerned that it will make this even more convoluted >"<